### PR TITLE
Modify bash script and action variables 

### LINKED ISSUE #60: Stability issues wrt Scheduled Benchmark Tests on Self Hosted Runner
#### Description

Multiple issues of the Scheduled Benchmark Tests on Self Hosted Runner generate different results

While here just one combination failed :- https://github.com/layer5io/meshery-smp-action/runs/7770247406 (linkerd-wrk-soak)

Here a couple failed :- https://github.com/layer5io/meshery-smp-action/actions/runs/2832186044

And here one runner startup itself failed :- https://github.com/layer5io/meshery-smp-action/actions/runs/2833335073

#### Expected Behavior

The behaviour should be consistent across Scheduled Benchmark Test run on self hosted runners. Given we are working with external hardware where connections can fail, ideally we should have a retry mechanism and we should work to considerably reduce the consistency

#### Logs
In the logs Auth seems to be a common culprit
```
Opening Meshery (http://localhost:31391) in browser.
Failed to open Meshery in browser, please point your browser to http://localhost:31391 to access Meshery.
authentication failed: Get "http://localhost:31[39](https://github.com/layer5io/meshery-smp-action/runs/7765562299?check_suite_focus=true#step:5:40)1/api/providers": dial tcp [::1]:31391: connect: connection refused
Verifying prerequisites...
Authentication token not found. please supply a valid user token with the --token (or -t) flag. or login with `mesheryctl system login`
Onboarding application... Standby for few minutes...
Error: Authentication token not found. please supply a valid user token with the --token (or -t) flag. or login with `mesheryctl system login`
```

This is seen for istio
```
Opening Meshery (http://192.168.49.2:32398/) in browser.
Failed to open Meshery in browser, please point your browser to http://192.168.49.2:32398/ to access Meshery.
Verifying prerequisites...
Adapter for required mesh not found
Onboarding application... Standby for few minutes...
rpc error: code = Unknown desc = no matches for kind "Gateway" in version "networking.istio.io/v1alpha3"

Error from server (NotFound): namespaces "istio-system" not found
Error from server (NotFound): namespaces "istio-system" not found
Service Mesh: Istio - ISTIO
Gateway URL: [http://192.168.49.2:](http://192.168.49.2/)
```

#### Environment:
- Meshery Version: v0.6.0-rc.6fc
- Kubernetes Version:  minikube 1.23.2
- Host OS: Ubuntu
- Browser: N/A

### PR DESCRIPTION
**Pull request by alphaX86**
Signed-off-by: Aadhitya A <aadhitya864@gmail.com>

**Description**

This PR fixes #60 (auth issues alone)

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by alphaX86** [comment]
Ok... I've changed some args in the script, seems now it works... at least for install section
image

**Comment by alphaX86** [comment]
Also, I've seen that `app onboard` now asks `-s` flag for source-type... Is this due to recent change?
image


// @theBeginner86

**Comment by theBeginner86** [comment]
> Also, I've seen that `app onboard` now asks `-s` flag for source-type (in Linkerd script)... Is this due to recent change? image
> 
> // @theBeginner86

Yeah, `mesheryctl` was recently updated in response to the effort for adding support for different source types for Applications on Meshery ([PR](https://github.com/meshery/meshery/pull/5775)). Since that Linkerd script uses `mesheryctl` so that is showing up.

File: .github/workflows/configurable-benchmark-test.yaml [hunk 1]
```diff
@@ -53,8 +53,8 @@ jobs:
           .github/workflows/scripts/${{ github.event.inputs.service_mesh }}_deploy.sh
         shell: bash
 
-      - name: Initiate tmate session
-        uses: mxschmitt/action-tmate@v3
+      #- name: Initiate tmate session
```
Comment (gyohuangxin) [inline .github/workflows/configurable-benchmark-test.yaml:56]: If it's no longer necessary, please delete the comment.

**Review by gyohuangxin — COMMENTED** [review]
Thanks for fixing it.

File: .github/workflows/configurable-benchmark-test.yaml (hunk 1 above)
Comment (alphaX86) [reply .github/workflows/configurable-benchmark-test.yaml:56]: Ok... I'll delete the action step

**Comment by gyohuangxin** [comment]
@alphaX86 Is it ok for review?

**Comment by alphaX86** [comment]
Yes @gyohuangxin ready for review.

// @hershd23

**Review by gyohuangxin — APPROVED** [review]
LGTM

File: .github/workflows/scripts/linkerd_deploy.sh [hunk 2]
```diff
@@ -16,10 +16,10 @@ fi
 
 curl -fsL https://run.linkerd.io/emojivoto.yml --output emojivoto.yml
 sleep 10
-mesheryctl system login --provider None
-mesheryctl mesh deploy adapter meshery-linkerd:10001
+#mesheryctl system login --provider None
+echo | mesheryctl mesh deploy adapter meshery-linkerd:10001 --token "./.github/workflows/auth.json"
 echo "Onboarding application... Standby for few minutes..."
-mesheryctl app onboard -f "./emojivoto.yml"
+mesheryctl app onboard -f "./emojivoto.yml" -s "Kubernetes Manifest" -t "./.github/workflows/auth.json"
```
Comment (hershd23) [inline .github/workflows/scripts/linkerd_deploy.sh:22]: Let's just use `--token` here just for consistency

File: .github/workflows/scripts/osm_deploy.sh [hunk 3]
```diff
@@ -16,10 +16,10 @@ fi
 
 curl -fsL https://raw.githubusercontent.com/openservicemesh/osm-docs/main/manifests/apps/bookstore.yaml --output bookstore.yaml
 sleep 10
-mesheryctl system login --provider None
-mesheryctl mesh deploy adapter meshery-osm:10009
+#mesheryctl system login --provider None
+echo | mesheryctl mesh deploy adapter meshery-osm:10009 --token "./.github/workflows/auth.json"
 echo "Onboarding application... Standby for few minutes..."
-mesheryctl app onboard -f "./bookstore.yaml"
+mesheryctl app onboard -f "./bookstore.yaml" -s "Kubernetes Manifest" -t "./.github/workflows/auth.json"
```
Comment (hershd23) [inline .github/workflows/scripts/osm_deploy.sh:22]: Same. Let's just use `--token` here just for consistency

**Review by hershd23 — COMMENTED** [review]
Just a couple of comments. Otherwise LGTM

**Comment by hershd23** [comment]
Also got to learn something cool. The entire file `meshery.sh` file seems to be in the git diff for a 1 line change. This is due to line endings. CRLF vs LF.

A great thread for anyone who wants to know more :- https://stackoverflow.com/questions/37344280/git-diff-is-showing-full-file-has-changed-for-a-single-line-change-but-only-for

**Comment by leecalcote** [comment]
I'm adding MeshMate @Nikhil-Ladha, who might speak to how we do linting and code conventions differently depending about the project.

File: .github/workflows/scripts/osm_deploy.sh (hunk 3 above)
Comment (leecalcote) [reply .github/workflows/scripts/osm_deploy.sh:22]:
```suggestion
mesheryctl app onboard -f "./bookstore.yaml" -s "Kubernetes Manifest" --token "./.github/workflows/auth.json"
```

File: .github/workflows/scripts/linkerd_deploy.sh (hunk 2 above)
Comment (leecalcote) [reply .github/workflows/scripts/linkerd_deploy.sh:22]:
```suggestion
mesheryctl app onboard -f "./emojivoto.yml" -s "Kubernetes Manifest" --token "./.github/workflows/auth.json"
```

**Review by leecalcote — APPROVED** [review]

**Comment by Nikhil-Ladha** [comment]
> Also got to learn something cool. The entire file `meshery.sh` file seems to be in the git diff for a 1 line change. This is due to line endings. CRLF vs LF.
> 
> A great thread for anyone who wants to know more :- https://stackoverflow.com/questions/37344280/git-diff-is-showing-full-file-has-changed-for-a-single-line-change-but-only-for
> 
> In the same vein @leecalcote do we enforce a certain type of line endings? Have seen some orgs being particular about it

Hmm....what type did you change to in this PR?
We should be following the LF format everywhere and git provides a way to do so by using .gitattributes file. Do check once, to enforce the LF line ending for all files.
Like we do for https://github.com/layer5io/layer5 repo.

**Comment by alphaX86** [comment]
> > Also got to learn something cool. The entire file `meshery.sh` file seems to be in the git diff for a 1 line change. This is due to line endings. CRLF vs LF.
> > 
> > A great thread for anyone who wants to know more :- https://stackoverflow.com/questions/37344280/git-diff-is-showing-full-file-has-changed-for-a-single-line-change-but-only-for
> > 
> > In the same vein @leecalcote do we enforce a certain type of line endings? Have seen some orgs being particular about it
> 
> Hmm....what type did you change to in this PR?
> We should be following the LF format everywhere and git provides a way to do so by using .gitattributes file. Do check once, to enforce the LF line ending for all files.
> Like we do for https://github.com/layer5io/layer5 repo.

So in short, you want me to revert the whole change in `meshery.sh` and then do the intended change I wanted to do.... Ok got it

**Comment by alphaX86** [comment]
Wait... reverting changes now...

**Comment by alphaX86** [comment]
Phew... I found a way to change line endings in Windows... almost it cost me up to change endings of whole repo... I reverted it so no issues. But it cost me a DCO check fail now... even though I sign-off now

**Review by hershd23 — APPROVED** [review]
LGTM

**Comment by alphaX86** [comment]
Umm... @leecalcote are we good to go?

**Review by gyohuangxin — APPROVED** [review]
LGTM

**Comment by gyohuangxin** [comment]
@alphaX86 Can you resolve the DCO issue?

**Comment by alphaX86** [comment]
If I resolve dco issue, it'd result in opening a new PR. Is it OK

**Comment by gyohuangxin** [comment]
@alphaX86 OK, I merged it. Let's see if it works.

**Comment by alphaX86** [comment]
> @alphaX86 OK, I merged it. Let's see if it works.

Let's hope ✌️

**Comment by gyohuangxin** [comment]
@alphaX86 Some performance tests still failed, https://github.com/layer5io/meshery-smp-action/runs/8062696783?check_suite_focus=true. Can you check if your changes worked?

**Pull request merged**